src/server/req_modify.c

```diff
@@ -743,6 +739,154 @@ is_select_smaller(char *select_orig, char *select_smaller)
 	return 1;
 }
 
+/**
+ * @brief Save the current starttime, duration, and select of a reservation.
+ * 		To be reverted by revert_alter_reservation
+ *
+ * @param[in] preq pointer to reservation
+ */
+void save_alter_reservation(resc_resv *presv) {
+
+	attribute *alter_revert;
+	resource_def *presdef = NULL;
+	resource *resc = NULL, *pselect = NULL;
+	attribute atemp = {0};
+
+	alter_revert = &presv->ri_wattr[RESV_ATR_alter_revert];
+	if (alter_revert->at_flags & ATR_VFLAG_SET)
+		return;
+
+	alter_revert->at_flags |= ATR_SET_MOD_MCACHE;
```

Comment:
We are trying to move away from setting attributes without calling the setter functions. So in this case, can we use set_resc() instead? You can probably make a list of new resources to add and then call set_resc()?

Reply:
is add_resource_entry + resource_def::rs_set not a valid way of doing this?
add_resource_entry does set alter_revert's at_flags to ATR_SET_MOD_MCACHE, so i can remove this line